app/models/ingest_post_processor.rb

```diff
@@ -0,0 +1,31 @@
+class IngestPostProcessor
+  attr_reader :params, :framework
+
+  def initialize(params:, framework:)
+    @params = params
+    @framework = framework
+  end
+
+  def resolve_parameters
+    params[:total_value] = total_value_from_data_hash
+    params
+  end
+
+  private
+
+  def total_value_from_data_hash
+    return if params[:data].blank?
+
+    params[:data][total_value_field]
+  end
+
+  def total_value_field
+    "#{framework.definition}::#{entry_type.capitalize}".constantize.total_value_field
```

Comment:
This is now the second place now we're doing this mapping of an entry type to the sheet definition for a framework and I imagine it will happen elsewhere too. how about we create a `Framework::Definition#sheet(entry_type)` method that returns either the `Invoice` or `Contract` class dependent on the entry type passed in?

Reply:
:+1: